docs/patterns/tools/use-tools/processing-actions.mdx

```diff
@@ -127,12 +96,112 @@ agent_executor = AgentExecutor(agent=agent, tools=tools, verbose=True)
 
 agent_executor.invoke({"input": task})
 ```
-```javascript JavaScript
-Coming Soon
+</Step>
+</Steps>
+</Tab>
+<Tab title="TypeScript">
+<Steps>
+<Step title="Import required libraries">
+```typescript TypeScript
+import { ActionExecutionResDto, LangchainToolSet, RawActionData, TPostProcessor, TPreProcessor, TSchemaProcessor } from "composio-core";
+import { ChatOpenAI } from "@langchain/openai";
+import { createOpenAIFunctionsAgent, AgentExecutor } from "langchain/agents";
+import { pull } from "langchain/hub";
+import { ChatPromptTemplate } from "@langchain/core/prompts";
+```
+</Step>
+<Step title="Initialize ChatOpenAI & LangChain toolset client">
+```typescript TypeScript
+const llm = new ChatOpenAI({ apiKey: "<your-api-key>" });
+const toolset = new LangchainToolSet({ apiKey: "<your-api-key>" });
+```
+</Step>
+<Step title="Schema Modifier">
+This will be used to modify the schema of the `LINEAR_CREATE_LINEAR_ISSUE` action, we remove the parameters `project_id` and `team_id` as required fields, later in the program we will pass these values as inputs to the action manually. The technical term for this is **Schema Processing**.
+```typescript TypeScript
+const schemaProcessor: TSchemaProcessor = ({
+  actionName,
+  toolSchema,
+}: {
+  actionName: string;
+  toolSchema: RawActionData;
+}) => {
+  const modifiedSchema = { ...toolSchema };
+  modifiedSchema.parameters = {
+    ...modifiedSchema.parameters,
+    required: modifiedSchema.parameters?.required?.filter(
+      field => !['project_id', 'team_id'].includes(field)
+    ) || []
+  };
+
+  return modifiedSchema;
+};
+```
+</Step><Step title="Input Modifier">
+This will be used to modify the input data for the `LINEAR_CREATE_LINEAR_ISSUE` action. Here we have added the values for `project_id` and `team_id` parameters to the input data. By doing this, we can avoid specifying these values in the prompt and be sure that the agent uses the correct values. The technical term for this is **Pre-Processing**.
+```typescript TypeScript
+const preProcessor: TPreProcessor = ({ params, actionName, appName }: {
+  params: Record<string, unknown>;
+  actionName: string;
+  appName: string;
+}) => {
+  const modifiedParams = { ...params };
+
+  modifiedParams.project_id = "e708162b-9b1a-4901-ab93-0f0149f9d805";
+  modifiedParams.team_id = "249ee4cc-7bbb-4ff1-adbe-d3ef2f3df94e";
+
+  return modifiedParams;
+}
+```
+</Step>
+<Step title="Output Modifier">
+This will be used to modify the output data for the `LINEAR_CREATE_LINEAR_ISSUE` action. Here we are modifying the output to just return the action execution status `successful` & the `issueId`, by doing this can keep the LLM context clean. The technical term for this is **Post-Processing**.
+```typescript TypeScript
+const postProcessor: TPostProcessor = ({ actionName, appName, toolResponse }: {
+  actionName: string;
+  appName: string;
+  toolResponse: ActionExecutionResDto;
+}) => {
+  const issueId = toolResponse.data.id;
```

Comment:
Consider adding error handling for the case when `toolResponse.data.id` is undefined:
```typescript
const postProcessor: TPostProcessor = ({ actionName, appName, toolResponse }: {
  actionName: string;
  appName: string;
  toolResponse: ActionExecutionResDto;
}) => {
  if (!toolResponse.data?.id) {
    return { data: null, successful: false, error: 'No issue ID found in response' };
  }
  const issueId = toolResponse.data.id;
  return { data: { id: issueId }, successful: true };
}